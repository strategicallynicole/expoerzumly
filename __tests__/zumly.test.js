/* eslint-env jest */
/*
https://github.com/mawrkus/js-unit-testing-guide
Here's how a typical test flow looks like:
-import the function to test
-give an input to the function
-define what to expect as the output
-check if the function produces the expected output

testear metodos de zumly

testear simulacion de interaccion (ver tobecalled)

testear util functions


Test simple user actions
Example of simple user actions:

Clicking on a link that toggles the visibility of a DOM element
Submitting a form that triggers the form validation
(...)

Note how simple the test is because the UI (DOM) layer does not mix with the business logic layer:

a "click" event occurs
a public method is called
The next step could be to test the business logic implemented in "showPreview()" or "hidePreview()".

*/
import {Zumly} from '../src/zumly.js'
import {jest} from '@jest/globals';
// some variables for testing purposes
var zumly0
var zumly1
var homeView


beforeAll(() => {
  // prepare template for testing
  document.body.innerHTML = `<!-- DIV canvas -->
    <div class="first zumly-canvas"></div>
    <div class="second zumly-canvas"></div>`;

  homeView = `<div class="z-view">
    <div class='card zoom-me' data-to='newView'></div>
  </div>`
      
  // prepare zumly instances for testing
  zumly0 = new Zumly({
        mount: '.first',
        initialView: 'homeView',
        views: {homeView}
      })
  zumly0.init()
  
  zumly1 = new Zumly({
        mount: '.second',
        initialView: 'homeView',
        views: {homeView},
        transitions: {
          effects: ['blur','sepia','saturate']
        }
      })
  zumly1.init()
});

describe('Zumly parameters', () => {

  describe('that are required', () => {

    it('mount should be a string', () => {

      expect(zumly0.mount).toBe('.first')

    })

    it('initialView should be a string', () => {

      expect(zumly0.initialView).toBe('homeView')
      
    })

    it('views should be an object', () => {

      expect(zumly0.views).toMatchObject({homeView})
      
    })

  })

  describe('that are optional', () => {

    it('cover should be a string', () => {

      expect(zumly0.cover).toBe('width')
      
    })

    it('duration should be a string', () => {

      expect(zumly0.duration).toBe('1s')
      
    })

    it('ease should be a string', () => {

      expect(zumly0.ease).toBe('ease-in-out')
      
    })

    it('effects should be an array. default', () => {

      expect(zumly0.effects).toEqual(expect.arrayContaining(['none','none']))
      
    })

    it('effects should be an array. custom', () => {

      expect(zumly1.effects).toEqual(expect.arrayContaining(['blur(0px) sepia(0) saturate(0) ', 'blur(0.8px) sepia(5) saturate(8) ']))
      
    })

    it('debug should be a boolean', () => {

      expect(zumly0.debug).toBe(false)
      
    })

  })

})

describe('Zumly instances', () => {

  it('first instance should be 1', () => {

    expect(zumly0.instance).toBe(1)

  })

  it('second instance should be 2', () => {

    expect(zumly1.instance).toBe(2)
    
  })

  it('should be reflected on DOM', () => {

     expect(document.body).toMatchSnapshot()
    
  })

})
  
 
