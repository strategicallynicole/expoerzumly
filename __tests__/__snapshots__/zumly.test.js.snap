// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zumly instances should be reflected on DOM 1`] = `
<body>
  <!-- DIV canvas -->
  
    
  <div
    class="first zumly-canvas"
    tabindex="0"
  >
    <div
      class="z-view is-current-view"
      data-view-name="homeView"
      style="transform-origin: 0 0;"
    >
      
    
      <div
        class="card zoom-me"
        data-to="newView"
      />
      
  
    </div>
  </div>
  
    
  <div
    class="second zumly-canvas"
    tabindex="0"
  >
    <div
      class="z-view is-current-view"
      data-view-name="homeView"
      style="transform-origin: 0 0;"
    >
      
    
      <div
        class="card zoom-me"
        data-to="newView"
      />
      
  
    </div>
  </div>
</body>
`;
